src/helper.rs

```diff
@@ -448,9 +472,37 @@ pub trait AuthenticatorDelegate {
     /// * Will only be called if the Authenticator's flow_type is `FlowType::Device`.
     fn present_user_code(&mut self, pi: &PollInformation) {
         println!("Please enter {} at {} and grant access to this application",
-                  pi.user_code, pi.verification_url);
+                 pi.user_code,
+                 pi.verification_url);
         println!("Do not close this application until you either denied or granted access.");
-        println!("You have time until {}.", pi.expires_at.with_timezone(&Local));
+        println!("You have time until {}.",
+                 pi.expires_at.with_timezone(&Local));
+    }
+
+    /// Only method currently used by the InstalledFlow.
+    /// We need the user to navigate to a URL using their browser and potentially paste back a code
+    /// (or maybe not). Whether they have to enter a code depends on the InstalledFlowReturnMethod
+    /// used.
+    fn present_user_url(&mut self, url: &String, need_code: bool) -> Option<String> {
+        if need_code {
+            println!("Please direct your browser to {}, follow the instructions and enter the \
+                      code displayed here: ",
+                     url);
+
+            let mut code = String::new();
+            let _ = io::stdin().read_line(&mut code);
```

Comment:
I am not sure what the advised style is, but what I am usually doing is this:
`io::stdin().read_line(&mut code).ok()`.